src/EventListener/RespondListener.php

```diff
@@ -42,15 +42,29 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
             return;
         }
 
+        $headers = [
+            'Content-Type' => sprintf('%s; charset=utf-8', $request->getMimeType($request->getRequestFormat())),
+            'Vary' => 'Accept',
+            'X-Content-Type-Options' => 'nosniff',
+            'X-Frame-Options' => 'deny',
+        ];
+
+        if (\in_array($request->getMethod(), ['POST', 'PUT', 'PATCH'])) {
+            if ($request->attributes->has('_api_write_item_iri')) {
+                $headers['Content-Location'] = $request->attributes->get('_api_write_item_iri');
+
+                if ($request->isMethod('POST')) {
+                    $headers['Location'] = $request->attributes->get('_api_write_item_iri');
+                }
+            } else {
+                @trigger_error(sprintf('No request attribute from `_api_write_item_iri` key is deprecated since API Platform 2.3 and will not be supported in API Platform 3, an string should always be returned. see deprecated into %s constructor for more details.', WriteListener::class), E_USER_DEPRECATED);
```

Comment:
Not sure if we should have this deprecation error here, but:

> Not setting the `_api_write_item_iri` request attribute is deprecated since API Platform 2.3 and will not be supported in API Platform 3. See %s for the default implementation.